gossip/src/cluster_info_duplicate_shred_listener.rs

```diff
@@ -0,0 +1,138 @@
+use {
+    crate::{
+        cluster_info::{ClusterInfo, GOSSIP_SLEEP_MILLIS},
+        crds::Cursor,
+        duplicate_shred::DuplicateShred,
+    },
+    log::*,
+    std::{
+        sync::{
+            atomic::{AtomicBool, Ordering},
+            Arc,
+        },
+        thread::{self, sleep, Builder, JoinHandle},
+        time::Duration,
+    },
+};
+
+pub trait ClusterInfoDuplicateShredHandler: Send {
+    fn handle(&mut self, data: DuplicateShred);
+}
+
+pub struct ClusterInfoDuplicateShredListener {
+    thread_hdls: Vec<JoinHandle<()>>,
+}
+
+// Right now we only need to process duplicate proof, in the future the receiver
+// should be a map from enum value to handlers.
+impl ClusterInfoDuplicateShredListener {
+    pub fn new(
+        exit: Arc<AtomicBool>,
+        cluster_info: Arc<ClusterInfo>,
+        handler: impl ClusterInfoDuplicateShredHandler + 'static,
```

Comment:
why does it need `+ 'static`?

Reply:
error[E0310]: the parameter type `impl ClusterInfoDuplicateShredHandler` may not live long enough
  --> gossip/src/cluster_info_duplicate_shred_listener.rs:37:24
   |
37 |                   .spawn(move || {
   |  ________________________^
38 | |                     Self::recv_loop(exit, &cluster_info, handler);
39 | |                 })
   | |_________________^ ...so that the type `impl ClusterInfoDuplicateShredHandler` will meet its required lifetime bounds
   |
help: consider adding an explicit lifetime bound...
   |
32 |         handler: impl ClusterInfoDuplicateShredHandler + 'static,
   |                                                        +++++++++

Suggestions welcome